.github/workflows/master.yml

```diff
@@ -0,0 +1,123 @@
+name: Build and Release
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+env:
+  TAG_NAME: latest
+
+jobs:
+  buildroot:
+    name: Build Firmware
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - runcam_wifilink
+          - emax_wyvern-link
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Prepare firmware
+        run: |
+          echo "nameserver 8.8.8.8" | sudo tee /etc/resolv.conf
+          echo "CACHE_DATE=$(date +%m)" >> $GITHUB_ENV
+          sudo apt update
+          sudo apt install -y cpio rsync bc qemu-user-static binfmt-support
+
+      - name: Setup ccache
+        uses: actions/cache@v4
+        with:
+          path: /tmp/ccache
+          key: ${{ matrix.platform }}-${{ env.CACHE_DATE }}
+
+      - name: Build firmware
+        run: |
+          export GIT_HASH=$(git rev-parse --short $GITHUB_SHA)
+          export GIT_BRANCH=${GITHUB_REF_NAME}
+          echo "GIT_HASH=${GIT_HASH}" >> $GITHUB_ENV
+          echo "GIT_BRANCH=${GITHUB_BRANCH}" >> $GITHUB_ENV
+
+          mkdir -p /tmp/ccache
+          ln -s /tmp/ccache ${HOME}/.ccache
+
+          NAME=${{ matrix.platform }}
+          DEFCONFIG=${NAME}_defconfig bash build.sh
+
+      - name: Upload images to workflow artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.platform }}-images-${{ github.sha }}
+          path: |
+            output/${{ matrix.platform }}_defconfig/images/${{ matrix.platform }}_sdcard.img
+            output/${{ matrix.platform }}_defconfig/images/${{ matrix.platform }}_u-boot.bin
+            output/${{ matrix.platform }}_defconfig/images/${{ matrix.platform }}_emmc_bootloader.img
+            output/${{ matrix.platform }}_defconfig/images/${{ matrix.platform }}_rootfs.squashfs
+            output/${{ matrix.platform }}_defconfig/images/${{ matrix.platform }}.tar.gz
+          retention-days: 7
+
+  release-on-master:
+    name: Release Latest (on master merge)
+    runs-on: ubuntu-latest
+    needs: [buildroot]
+    # Only run when pushing to master (not on PRs) and not on tags
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master' && !startsWith(github.ref, 'refs/tags/')
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Download all artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: artifacts/
+
+      - name: Create/Update Latest Release
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: latest
```

Comment:
I know the idea of updating the `latest` tag is used by many OpenIPC projects, but it is very inconvenient because

* The `latest` tag sinks down on the releases page as soon as new versioned release happens
* If you decide to change the names of the artifacts, you'd have to remove all the old ones manually, because `overwrite:true` I believe appends and overrides the existing files, but doesn't remove the ones which are no longer actual

I think I saw sometimes ago projects which drop and re-create the `latest` release, but can't find a good example.